util/azure/azure.go

```diff
@@ -90,4 +90,23 @@ func ConvertResourceGroupNameToLower(resourceID string) (string, error) {
 
 	resourceGroup := matches[1]
 	return strings.Replace(resourceID, resourceGroup, strings.ToLower(resourceGroup), 1), nil
-}
\ No newline at end of file
+}
+
+// regionMapping contains mappings from custom region names to standard Azure region names.
+// This is useful for environments where custom region names are used that need to be
+// translated to standard Azure region names for API calls.
+var regionMapping = map[string]string{
+	"ussec": "eastus2",
+	// Add other custom region mappings as needed
+}
+
+// NormalizeAzureRegion translates custom region names to standard Azure region names.
+// If the region is not in the mapping, it returns the original region name.
+// This function is useful for environments like Azure Stack Edge, Azure Government,
+// or other deployments where custom region names are used.
+func NormalizeAzureRegion(region string) string {
```

Comment:
@HigashikataZhangsuke, this will only work for the emulated Sequoia case and will break for the actual Azure Secret cloud. Let's make it handle the non-emulated case too.